libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+    @Test
+    public void NonNullTelemetryClientSucceeds() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+
+        BotTelemetryClient botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void OverrideTest() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+        MyBotTelemetryClient botTelemetryClient = new MyBotTelemetryClient(telemetryClient);
+    }
+
+    @Test
+    public void TrackAvailabilityTest()
+    {
+        Map<String, String> properties = new HashMap<>();
+        Map<String, Double> metrics = new HashMap<>();
+        properties.put("hello", "value");
+        metrics.put("metric", 0.6);
+
+        botTelemetryClient.trackAvailability(
+            "test",
+            OffsetDateTime.now(),
+            Duration.ofSeconds(1000), // TODO: use computer ticks
+            "run location",
+            true,
+            "message",
+            properties,
+            metrics);
+
+        TelemetryChannel telemetryClient = Mockito.mock(TelemetryChannel.class);
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry eventName = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventName.getName());
+            Assert.assertEquals("value", eventName.getProperties().get("hello"));
+            Assert.assertEquals("0.6", eventName.getMetrics().get("metric"));
+
+            return null;
+        }).when(telemetryClient).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void TrackEventTest()
+    {
+        botTelemetryClient.trackEvent("test", new HashMap<String, String>() {{ put("hello", "value"); }}, new HashMap<String, Double>() {{ put("metric", 0.6); }});
+
+        TelemetryChannel telemetryClient = Mockito.mock(TelemetryChannel.class);
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry eventName = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventName.getName());
+            Assert.assertEquals("value", eventName.getProperties().get("hello"));
+            Assert.assertEquals("0.6", eventName.getMetrics().get("metric"));
+
+            return null;
+        }).when(telemetryClient).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void TrackDependencyTest()
+    {
+        botTelemetryClient.trackDependency("test", "target", "dependencyname", "data", OffsetDateTime.now(), Duration.ofSeconds(1000), "result", false);
+
+        TelemetryChannel telemetryClient = Mockito.mock(TelemetryChannel.class);
+        Mockito.doAnswer(invocation -> {
+            RemoteDependencyTelemetry eventName = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventName.getVer());
+            Assert.assertEquals("target", eventName.getTarget());
+            Assert.assertEquals("dependencyname", eventName.getName());
+            // Assert.assertEquals("data", eventName.getData());
```

Comment:
this is ok?

Reply:
No, that is not ok.
I wanted to analyze that because the class Java does not have a `getData()` method, its private.

Reply:
We can add a TODO here explaining why this is commented. +1 @VictorGrycuk

Reply:
Removed the comment, there doesnt seem to be a way to retrieve this in java